src/js/modules/articles/reducer.js

```diff
@@ -2,123 +2,83 @@ import {
-      volume: 108,
-      issue: 1,
-      date: "July 19, 2017",
-      time: "8:27 AM",
-      sectionSlug: "creative-writing",
-    },
-  },
+  articles: {},
 };
 
 const reducer = (state = { ...initialState }, action) => {
   switch (action.type) {
     case FETCH_ARTICLE_PENDING: {
-      return { ...state, isFetching: true };
-    }
-    case FETCH_ARTICLE_REJECTED: {
-      return { ...state, isFetching: false, error: action.payload };
+      return { ...state, isFetching: true, };
     }
     case FETCH_ARTICLE_FULFILLED: {
       return {
         ...state,
         isFetching: false,
         isFetched: true,
-        request: action.payload,
+        // action.payload already tested in actions to be array
+        articles: action.payload.reduce((acc, cur) => {
+          const articleSlug = cur.slug;
+          cur.dateline = formatDate(cur.created_at);
+          cur.sectionSlug = slugFinder(cur.section_id, action);
+          delete cur.section_id;
+          acc[ articleSlug ] = cur;
+          return acc;
+        }, { ...state.articles }),
+      };
+    }
+    case FETCH_ARTICLE_REJECTED: {
+      return {
+        ...state,
+        isFetching: false,
+        error: action.payload,
       };
     }
   }
   return state;
 };
 
-export default reducer;
+const formatDate = (createdAt) => {

```

Comment:
this `formatDate` function can be used for general use, so you shouldn't assume that the parameter will always be `createdAt`. Instead, just name it `string` or something.